### PR TITLE
Dictionary expressions: add resolutions for open questions from LDM-2024-03-11

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: proposals/dictionary-expressions.md [hunk 1]
```diff
@@ -149,6 +155,7 @@ Which approach should we go with with our dictionary expressions? Options includ
 2. Purely permissive.  All elements are added using the indexer.  Perhaps with compiler warnings if the exact same key is given the same constant value twice.
 3. Perhaps a hybrid model.  `.Add` if only using `k:v` and switching to indexers if using spread elements.  There is deep potential for confusion here.
 
+**Resolution:** Use *indexer* as the lowering form. [LDM-2024-03-11](https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-03-11.md#conclusions)
```
Comment (jnm2) [inline proposals/dictionary-expressions.md:158]: How should we handle the contradiction between this and the resolution to open question 3:

> *Dictionary elements* will be supported in collection expressions for collection types that have a key-value pair element type.

Which used this example, for which `list["mads"] = 21` will not work:

> `List<KeyValuePair<string, int>> list = ["mads": 21];`

Comment (cston) [reply proposals/dictionary-expressions.md:158]: Good catch. Added an open question.

Comment (CyrusNajmabadi) [reply proposals/dictionary-expressions.md:158]: godo question.  my preference is '3'.  normal pre-dictionary collection initializable types have 'Add' semantics for their elements.  I would prefer to stick with that for sanity/consistency.

Comment (jnm2) [reply proposals/dictionary-expressions.md:158]: Credit to Fred :)

**Pull request merged**